Crispresso submission takes range as two columns

RasVandoros opened the pull request:
Changed the handling of data in the crispresso submission script.
Instead of expecting a column called range that looks like: "1-96", it
now expects two separate columns that hold the minimum and the maximum indexes.

RasVandoros committed:
Crispresso submission takes range as two columns

Changed the handling of data in the crispresso submission script.
Instead of expecting a column called range that looks like: "1-96", it
now expects two seperate columns that hold the minimum and the maximum indexes.

RasVandoros committed:
Bsub script now works with min max and range

Works with text csv.

RasVandoros committed:
Finished project needs testing

Everything seems to be working so far. Needs further testing to make
sure that the bsub and efficiency calculator work as intended with
either range or min/max

RasVandoros committed:
Necessary commit to work on a different project

RasVandoros committed:
Now identifies properly if nhej/total exists

Instead of counting columns to identify if the frameshift calculator has
been run already, the script now greps for nhej column name. Perlcritic
was also run.

PeterFCKeen commented on bin/bsub_crispresso_jobs.pl:
```diff
@@ -30,74 +31,94 @@
+if ($col[5] =~  m/^min/gmi){
+    my @heads = qw(experiment gene crispr strand amplicon min_index max_index nhej total hdr);
+    $csv->column_names(\@heads);
+} else{
```
Space

PeterFCKeen commented on bin/bsub_crispresso_jobs.pl:
```diff
@@ -30,74 +31,94 @@
+}
+
+while (my $hr = $csv->getline_hr($fh)){
+    foreach my $key( keys %{ $hr } ){
```
map keys through lc into hash

PeterFCKeen commented on bin/bsub_crispresso_jobs.pl:
```diff
@@ -30,74 +31,94 @@
+        die;
+    }
+    my $crispr_site = substr($hr->{crispr},0,20);
+	foreach my $barcode (@barcodes) {
```
indentation

PeterFCKeen commented on bin/crispresso_nhej_counts_post-migration.pl:
```diff
@@ -184,24 +177,23 @@ sub well_builder {
+    my $column_number = scalar(@$header);
+    unless ( grep( m/NHEJ/gmi, @$header)) {
+        splice @$header, $column_number , 0, "NHEJ";
+        splice @$header, ($column_number+1), 0, "Total";
```
spacing

PeterFCKeen commented on bin/crispresso_nhej_counts_post-migration.pl:
```diff
@@ -184,24 +177,23 @@ sub well_builder {
-        splice @$header, 6, 0, "NHEJ";
-        splice @$header, 7, 0, "Total";
+    my $column_number = scalar(@$header);
+    unless ( grep( m/NHEJ/gmi, @$header)) {
```
spacing

RasVandoros committed:
Post Review Changes

Took out a redundant bit where I was lowercasing a hash. Took out NHEJ
and Total from the headers name, since it is not yet there in summary
file and therefore its just undef in every case.

RasVandoros committed:
Enabled multiple criprs
